tests/ChannelTest.php

```diff
@@ -2,117 +2,46 @@
-use Illuminate\Notifications\Notifiable;
-use Illuminate\Notifications\Notification;
 use Mockery;
 use NotificationChannels\ExpoPushNotifications\ExpoChannel;
-use NotificationChannels\ExpoPushNotifications\ExpoMessage;
+use NotificationChannels\ExpoPushNotifications\Test\Support\ChannelMockTrait;
+use NotificationChannels\ExpoPushNotifications\Test\Support\Notifications\TestNotifiable;
+use NotificationChannels\ExpoPushNotifications\Test\Support\Notifications\TestNotification;
 
 class ChannelTest extends TestCase
 {
-    /**
-     * @var Expo
-     */
-    protected $expo;
+    use ChannelMockTrait;
 
-    /**
-     * @var Dispatcher
-     */
-    protected $events;
-
-    /**
-     * @var ExpoChannel
-     */
-    protected $channel;
-
-    /**
-     * @var TestNotification
-     */
-    protected $notification;
-
-    /**
-     * @var TestNotifiable
-     */
-    protected $notifiable;
+    protected TestNotifiable $notifiable;
+    protected Expo $expoMock;
+    protected Dispatcher $events;
 
     protected function setUp(): void
     {
         parent::setUp();
 
-        $this->expo = Mockery::mock(Expo::class);
-
+        $this->notification = new TestNotification();
+        $this->expoMock = Mockery::mock(Expo::class);
         $this->events = Mockery::mock(Dispatcher::class);
-
-        $this->channel = new ExpoChannel($this->expo, $this->events);
-
-        $this->notification = new TestNotification;
-
-        $this->notifiable = new TestNotifiable;
-    }
-
-    protected function tearDown(): void
-    {
-        parent::tearDown();
-
-        Mockery::close();
-    }
-
-    // TODO: remove after checking the reason of missing assert error
-    /** @test */
-    public function itCanSendANotification()
-    {
-        $message = $this->notification->toExpoPush($this->notifiable);
-
-        $data = $message->toArray();
-
-        $this->expo->shouldReceive('notify')->with(['interest_name'], $data, true)->andReturn([['status' => 'ok']]);
-
-        $this->channel->send($this->notifiable, $this->notification);
-
-        $this->assertTrue(true);
     }
 
-    // TODO: remove after checking the reason of missing assert error
-    /** @test */
-    public function itFiresFailureEventOnFailure()
+    public function testItCanSendANotification()
     {
-        $message = $this->notification->toExpoPush($this->notifiable);
-
-        $data = $message->toArray();
+        $this->notification->toExpoPush(new TestNotifiable());
 
-        $this->expo->shouldReceive('notify')->with(['interest_name'], $data, true)->andThrow(ExpoException::class, '');
+        $this->assertNotificationSend('channel_message.json');
 
-        $this->events->shouldReceive('dispatch')->with(Mockery::type(NotificationFailed::class));
-
-        $this->channel->send($this->notifiable, $this->notification);
-
-        $this->assertTrue(true);
+        (new ExpoChannel($this->expoMock, $this->events))->send(new TestNotifiable(), new TestNotification());
```

Comment:
could we move `new ExpoChannel($this->expoMock, $this->events)` to the class property?